aspnetcore/blazor/Security-guidance.md

```diff
@@ -0,0 +1,254 @@
+  * When we force the UI to render all the todo items, the server is forced to hold on to additional memory for things that might not be visible in the screen at the time. (This assumes you don't render all the items and have some sort of paging)
+    * We recommend that you only render the components/data that needs to be visible to the user and avoid holding on to any other data in memory, specially if this data can grow based on user input like our `TodoList`. In this particular case you could achieve this by implementing some sort of _virtual-list_ that only exposes a window to the subset of elements at the current position/page for the component to render and keeps the rest either in secondary storage (ideal) or in memory (less ideal).
+
+While Blazor server-side applications offer a very similar programming model to other UI frameworks for stateful applications like WPF, Windows forms or Blazor client-side, the main thing to remember is that in all those cases the memory consumed by the application belongs to the client and only affects that individual client, while in the server case the memory consumed by the application belongs to the server and is shared with all the clients running on that server instance.
+
+This is true for all server-side applications, but most applications are stateless and the memory gets released at the end of the request.
+
+As a general recomendation, you should pay the same attention to the memory consumed by a given client as in any other server-side app, paying special attention to the fact that the memory consumed by a blazor server-side application lives for a much longer time than a simple request.
+
+### Connections
+Connection exhaustion can happen when a client opens too many concurrent connections to the server preventing new connections from other clients from getting established.
+
+Blazor clients establish a single connection per session and keep it open for as long as the browser window is opened, this however doesn't prevent other clients from opening multiple conections to the server. This is not specific to blazor applications but given the long-lived nature of connections in blazor applications and the stateful nature of the applications, this can have a bigger impact in terms of availability.
+
+This is not new to blazor server-side applications, other server-side applications have this concern too. We recommend two ways of dealing with this.
+* Limit the amount of established connections from a given client as you would do in any other application through the use of a proxy server. Azure offers several options for this like Azure Front Door.
+* Proxy websocket connections to your application through the use of a proxy like Azure SignalR Service that multiplexes the connections from clients to your application. This gives your application way more connections that what a single client can establish, preventing a client from exhausting the connections to the server.
```

Comment:
Maybe we should put the Azure SignalR advice first, and in fact emphasize that we recommend this as **the** solution for all Azure-deployed Blazor server apps. As in, we don't recommend running in production without it.

If people are using Azure SignalR, they have no need for the other bullet-point (Azure Front Door), right?